src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -90,6 +92,20 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  dout(20) << __func__ << ": " << "sending NBD_DISCONNECT" << dendl;
+  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
+    derr << "rbd-nbd: disconnect failed" << cpp_error(errno) << std::endl;
+  } else {
+    dout(20) << __func__ << ": " << "disconnected" << dendl;
+  }
```

Comment:
(1). Just took a simple look at the kernel code about nbd, at least we will get ```-EPERM```, ```-EINVAL``` or ```-EIO``` in ioctl(NBD_DISCONNECT). I think only in -EINVAL case, ```the device is not used``` is a good error message. Why we error this message out immediately without any checking?

(2) yes, signal handler is not called by user, but that does not mean it will never fail. Please consider this scenario,
```
Thread 1                                Thread 2
do map nbd A
recieve SIGINT
                                    disconnect nbd A
disconnect in sig_handler

```
Although this is a very rare case, there is no lock on this nbd device between map and disconnect. 


So I am still not convinced they should be different cases.